### PR TITLE
feat(call_decoder): Implement Call decoder functionality

### PR DESCRIPTION
**Pull request by SozinM**

### CODE REVIEW

File: src/decode_call.rs
Comment (JonoPrest) [inline src/decode_call.rs:34]: I personally prefer this to be passed in on construction than to be set afterwards, what do you think?

Comment (SozinM) [reply src/decode_call.rs:34]: We have the same logic for Decoder everywhere too
I don't have a strong opinion about that but changing it in Decoder will be breaking change, and having a different interface in CallDecoder would be weird 
We could try to accommodate both options.

Comment (JonoPrest) [reply src/decode_call.rs:34]: Yeah I've seen it, but it gets passed by config for instantiating the client. So both patterns currently exist. I would advocate for moving it to setting on construction everywhere but I understand that's not the scope of this and it's not blocking to getting this merged.

**Review by JonoPrest — APPROVED** [review]
LGTM @SozinM, I don't typically review PRs on this repo so I'll leave it to you if you want any other feedback before merging.

I would still prefer to add an API for whether or not to checksum on construction but this can be done at a later stage.

File: src/decode_call.rs
Comment (SozinM) [reply src/decode_call.rs:34]: Actually even changing the current interface (let's say we put it in from_sigs will break compatibility)

Comment (ozgrakkurt) [reply src/decode_call.rs:34]: We can add another constructor function like from_config_and_sigs() which takes config struct as well an has only one parameter for now. This won't be a breaking change. Can do it in next PR

**Review by ozgrakkurt — APPROVED** [review]

**Pull request merged**
